lib/fetch-playable-url.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+const Promise = require('bluebird');
+const request = require('request');
+
+const BASE_URL = 'http://player.ooyala.com/sas/player_api/v1/authorization/embed_code';
+const QUERY = 'device=roku&domain=www.ooyala.com&supportedFormats=m3u8';
+
+// args.apiKey
+// args.embedCode
+//
+// If the playable URL is not defined, this will reject with:
+//   error.code === 'STREAM_UNDEFINED'
+function fetchPlayableUrl(args) {
+	args = args || {};
+	const apiKey = args.apiKey;
+	const embedCode = args.embedCode;
+
+	if (!apiKey) {
+		throw new Error('args.apiKey is required');
+	}
+	if (!embedCode) {
+		throw new Error('args.embedCode is required');
+	}
+
+	const params = {
+		method: 'GET',
+		url: `${BASE_URL}/${apiKey}/${embedCode}?${QUERY}`
+	};
+
+	return fetchPlayableUrl.makeRequest(params).then(data => {
+		const stream = (data.authorization_data[embedCode].streams || [])[0];
```

Comment:
You may want to test what happens here when `authorization_data` is absent.  There is a possibility of errors coming back from this endpoint with a statusCode of 200.